### ISSUE TITLE
either as middleware or let configure a "client" prefix

### ISSUE THREAD

**Issue by simllll** [coderaiser/cloudcmd#38]
hi, 
i'm trying to achieve following:
http://host:port/**individual-prefix/**

right now, I only can host in in the webroot (e.g. http://host:port**/**) directory, as soon as i try to add a prefix everysthing breaks (e.g. by routing through nginx with a different location url http://externalurl/mycloudcmd/ -> http://localhost:8000/ ).

maybe I need to set this "client prefix" parameter somewhere? Event though I found "prefix" in the code, I couldn't figure out how to use it. As far as I've seen it, this "prefix" is something else and has not directly anything todo with the client prefix thingy I'm looking for.

So I see two possible solutions (either one would me make happy ;))
- either provide a config option to set a client prefix
- allow to use cloudcmd as middleware..

thanks,
looking forward to any kind of feedback :)

regards
simon

**Comment by coderaiser** [comment]
Hi, 
I had started work on this feature some time ago, it is not ready for now.
All main things located in files: [cloudcmd.js](https://github.com/coderaiser/cloudcmd/blob/b9a0e4e409d54be4f2ea0a84d3d568dd09e62335/lib/cloudcmd.js) and [index.html](https://github.com/coderaiser/cloudcmd/blob/ae75d3449f01674aad0dd71380361a9feab79a43/html/index.html).
`client prefix` it's a good idea.

I'll think about it :).

**Comment by coderaiser** [comment]
With help of https://github.com/coderaiser/cloudcmd/commit/a74e49f4ab6b8aec7f09f012bf557b350a36c65b you could do next things:

```
mkdir node_modules && cd node_modules
git clone https://github.com/coderaiser/cloudcmd.git
cd ..
npm i express socket.io
```
And write down `app.js`:
```js
var cloudcmd    = require('cloudcmd'),
    http        = require('http'),
    express     = require('express'),
    io          = require('socket.io'),
    app         = express(),
    socket,
    server;

server = http.createServer(app);
socket = io.listen(server);

app.use(cloudcmd({
    prefix: '/any-prefix',
    socket: socket
}));

server.listen(3000);
```
Then run `node app` and open browser with `http://localhost:3000/any-prefix`.

**Comment by simllll** [comment]
wow :) that was quick.. I give it a try right now! thanks man

**Comment by simllll** [comment]
you are aweseome :+1: 

thanks, works great!

**Comment by coderaiser** [comment]
Landed in [v3.4.0](https://github.com/coderaiser/cloudcmd/releases/tag/v3.4.0).

**Comment by coderaiser** [comment]
[Added ability](https://github.com/coderaiser/cloudcmd/commit/d62049cd5915681a1fe94e8901b442bf5cc645d0) to set `prefix` with `--prefix` command line parameter.

**Comment by coderaiser** [comment]
Ability to set `--prefix` landed in [v4.3.0](https://github.com/coderaiser/cloudcmd/releases/tag/v4.3.0)
